mono/mini/mini-runtime.c

```diff
@@ -2385,10 +2387,13 @@ mono_llvmonly_runtime_invoke (MonoMethod *method, RuntimeInvokeInfo *info, void
 
 	runtime_invoke (NULL, args, exc, info->compiled_method);
 
-	if (sig->ret->type != MONO_TYPE_VOID && info->ret_box_class)
-		return mono_value_box (domain, info->ret_box_class, retval);
-	else
+	if (sig->ret->type == MONO_TYPE_VOID || !info->ret_box_class)
 		return *(MonoObject**)retval;
+
+	box = mono_value_box_checked (domain, info->ret_box_class, retval, &error);
+	mono_error_raise_exception (&error); /* FIXME don't raise here */
```

Comment:
It changes the current behavior, and as soon as we merge https://github.com/mono/mono/pull/2505, we will not have to raise this MonoError here.